### PR TITLE
(#1063) All classes implementing Text need to be refactored to extend TextEnvelope

### PR DESCRIPTION
**Pull request by Umbrah**
This is for #1063 

I replaced implementing Text with extending TextEnvelope and moved logic from asString to constructor.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1108](https://github.com/yegor256/cactoos/issues/1108) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1108](https://github.com/yegor256/cactoos/issues/1108) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/33dd77b0-00e8-4e63-ad6b-5950ef39e2ab); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by Umbrah** [comment]
@victornoel ping

**Comment by victornoel** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/1108#issuecomment-484925986))

@victornoel The user @victornoel[/z](https://www.0crat.com/u/victornoel) resigned from [#1108](https://github.com/yegor256/cactoos/issues/1108), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
This pull request [#1108](https://github.com/yegor256/cactoos/issues/1108) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/fef932c0-761d-41dc-8906-e0174aedf62c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by Umbrah** [comment]
@ilyakharlamov ping

File: src/main/java/org/cactoos/text/Abbreviated.java [hunk 1]
```diff
@@ -31,11 +32,8 @@
  * <p>There is no thread-safety guarantee.
  *
  * @since 0.29
- * @todo #897:30min All classes implementing Text need to be refactored
```
[the remark this answers is not in the document]
Comment (Umbrah) [reply src/main/java/org/cactoos/text/Abbreviated.java]: @ilyakharlamov I added to-do to another class which needs refactoring, and added a note there that this to-do should be moved to another one after finish

[the remark this answers is not in the document]
Comment (ilyakharlamov) [reply src/main/java/org/cactoos/text/Abbreviated.java]: thanks

**Comment by ilyakharlamov** [comment]
@Umbrah rebase into a singular, atomic, nice looking change and we are good to go

**Comment by Umbrah** [comment]
@ilyakharlamov Done

**Comment by ilyakharlamov** [comment]
@llorllale it's good to merge

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1108](https://github.com/yegor256/cactoos/issues/1108) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@ilyakharlamov please make sure that next time during the code review you will try to find at least 3 issues as written in [policy](http://www.zerocracy.com/policy.html#30)

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request merged**